pkg/cloud/services/vpc/service.go

```diff
@@ -213,6 +223,45 @@ func (s *Service) GetVPCByName(vpcName string) (*vpcv1.VPC, error) {
 	return vpc, nil
 }
 
+// GetImageByName returns the VPC Custom Image with given name. If not found, returns nil.
+func (s *Service) GetImageByName(imageName string) (*vpcv1.Image, error) {
+	var image *vpcv1.Image
+	f := func(start string) (bool, string, error) {
+		// check for existing images
+		listImagesOptions := &vpcv1.ListImagesOptions{}
+		if start != "" {
+			listImagesOptions.Start = &start
+		}
+
+		imagesList, _, err := s.ListImages(listImagesOptions)
+		if err != nil {
+			return false, "", err
+		}
+
+		if imagesList == nil {
+			return false, "", fmt.Errorf("image list returned is nil")
+		}
+
+		for i, v := range imagesList.Images {
+			if (*v.Name) == imageName {
```

Comment:
```suggestion
			if *v.Name == imageName {
```

Reply:
Sure, then I'd recommend the other `Get*ByName` functions get updated later as well.